ml_scraping/src/RF_model.py

```diff
@@ -0,0 +1,174 @@
+for r in posShort.split("\n"):
+    documents.append( (r, 0) )
+    words = word_tokenize(r)
+    pos = nltk.pos_tag(words)
+    for w in pos:
+        if w[1][0] in allowed_word_types:
+            all_words.append(lemmatizer.lemmatize( w[0].lower()) )
+#neg_reviews = [r for r in neg.split("\n")]
+#pos_reviews = [r for r in pos.split("\n")]
+
+for r in negMovie:
+    documents.append( (r, 1) )
+    words = word_tokenize(r)
+    pos = nltk.pos_tag(words)
+    for w in pos:
+        if w[1][0] in allowed_word_types:
+            all_words.append(lemmatizer.lemmatize( w[0].lower()) )
+
+for r in posMovie:
+    documents.append( (r, 0) )
+    words = word_tokenize(r)
+    pos = nltk.pos_tag(words)
+    for w in pos:
+        if w[1][0] in allowed_word_types:
+            all_words.append(lemmatizer.lemmatize( w[0].lower()) )
+    
+
+# save_word_features = open(glob.glob(os.path.join(DATA_DIR,"pickled_algos", "word_features20%.pickle","wb")
+# pickle.dump(word_features, save_word_features)
+# save_word_features.close()
+'''
+[pos_reviews.append(r) for r in posMovie]
+[neg_reviews.append(r) for r in negMovie]
+
+# tokenize pos & neg reviews
+tokenized_pos = [(word_tokenize(r)) for r in pos_reviews]
+tokenized_neg = [(word_tokenize(r)) for r in neg_reviews]
+pos_pos = nltk.pos_tag(tokenized_pos)
+pos_neg = nltk.pos_tag(tokenized_neg)
+
+
+
+# remove stop words
+stop_words = set(stopwords.words("english"))
+lemmatizer = WordNetLemmatizer()
+
+# lemmatize pos & neg reviews
+lematized_pos = [[lemmatizer.lemmatize(word) for word in sentence if word not in stop_words and word not in punctuation] for sentence in pos_pos]
+lematized_neg = [[lemmatizer.lemmatize(word) for word in sentence if word not in stop_words and word not in punctuation] for sentence in pos_neg]
+#print(lematized_neg[:2])
+'''
+# bag of words all data
+# 20%
+'''
+arr_one = np.array(lematized_pos).flatten()
+arr_two = np.array(lematized_neg).flatten()
+combined = []
+def add(text):
+    if type(text) != str:
+        for l in text:
+            add(l)
+    else:
+        combined.append(text)
+
+add(arr_one)
+add(arr_two)
+'''
+# number of unique words
+all_words = nltk.FreqDist(all_words)
+word_features = list( all_words.keys() )[:len(all_words.keys())//5 ]
+# save_word_features = open(os.path.join(DATA_DIR, "picked_algos", "WordFeatures.pickle"),"wb")
+# pickle.dump(word_features, save_word_features)
+# save_word_features.close()
+def find_features(document):
+    words = word_tokenize(document)
+    lem_words = [lemmatizer.lemmatize(w) for w in words]
+    features = {}
+    for w in word_features:
+        features[w] = (w in lem_words)
+
+    return features
+
+featuresets = [(find_features(rev), category) for (rev, category) in documents]
+ 
+random.shuffle(featuresets)
+print(len(featuresets))
+testing_set = featuresets[len(featuresets)//100:]
+training_set = featuresets[:len(featuresets)//100]
+
+x_train = [ [i[0][w] for w in word_features] for i in training_set]  
```

Comment:
you only need one list comprehension for this, because the first of the tuple is going to be a list of features
```python
x_train = [x[0] for x in training_set]
```